internal/process/deprovisioning/check_kyma_resource_deleted_step.go

```diff
@@ -62,16 +62,15 @@ func (step *CheckKymaResourceDeletedStep) Run(operation internal.Operation, logg
 		Name:      kymaResourceName,
 	}, kymaUnstructured)
 
-	if err == nil {
-		logger.Infof("Kyma resource still exists")
-		return step.operationManager.RetryOperationWithoutFail(operation, step.Name(), "Kyma resource still exists", 5*time.Second, step.kymaResourceDeletionTimeout, logger, nil)
-	}
-
-	if !errors.IsNotFound(err) {
+	if !errors.IsNotFound(err) && err != nil {
```

Comment:
How isNotFound handles err = nil
Should not we change the order?
```suggestion
	if err != nil && !errors.IsNotFound(err) {
```

Reply:
When err is nil then IsNotFound returns false.